uint8 LEFT=0
uint8 MIDDLE=1
uint8 RIGHT=2
uint8 m_type
